chore: rename advanced permissions to access control

zlwaterfield opened the pull request:
## Changes

Renaming advanced permissions to access control. Note: this only renames the label. It also access the access control feature. It's too hard to update an existing feature entitlement key so we will do a switch over after adding it billing and doing a full sync. 

See billing side: https://github.com/PostHog/billing/pull/1440.

## How did you test this code?

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

<!-- Docs reminder: If this change requires updated docs, please do that! Engineers are the primary people responsible for their documentation. 🙌 -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

zlwaterfield committed:
Rename 'Advanced permissions' to 'Access control'

Updated references and display names from 'Advanced permissions' to 'Access control' in billing fixtures and a related comment to ensure consistent terminology across the application.

zlwaterfield committed:
Add 'access_control' feature to billing and types

Introduces the 'access_control' feature to billing fixtures, TypeScript enums, and Python constants. This prepares the codebase for future migration away from 'advanced_permissions' and ensures consistent feature flagging across frontend and backend.

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

github-actions[bot] committed:
Update UI snapshots for `chromium` (9)

github-actions[bot] committed:
Update UI snapshots for `chromium` (10)

github-actions[bot] committed:
Update UI snapshots for `chromium` (11)

pawel-cebula reviewed (approved):
Fixture changes will be included in the follow up PRs?

zlwaterfield committed:
Add 'access_control' feature to billing fixtures

Introduces the 'access_control' feature to various billing-related JSON fixtures for Cypress, Playwright, and frontend mocks. Also removes the 'Project-based only' note from 'advanced_permissions' and updates the PayGateMini story to remove an unused limit property. This ensures consistency and prepares the test data for new or updated billing features.

zlwaterfield commented:
Added!

github-actions[bot] committed:
Update UI snapshots for `chromium` (10)
